code/renderers/react/src/renderToCanvas.tsx

```diff
@@ -81,12 +98,22 @@ export async function renderToCanvas(
     unmountElement(canvasElement);
   }
 
-  await act(async () => {
-    await renderElement(element, canvasElement, storyContext?.parameters?.react?.rootOptions);
+  await new Promise<void>((resolve, reject) => {
+    try {
+      actQueue.push(async () => {
+        await act(async () => {
+          await renderElement(element, canvasElement, storyContext?.parameters?.react?.rootOptions);
+          resolve();
+        });
+      });
+      processActQueue();
```

Comment:
I think you need to `await` or `.catch()` here, otherwise any errors thrown from the act will be uncaught.

Reply:
Awaiting `processActQueue` would lead to the whole queue processing before continuing a specific render. Also, catching here and calling `reject(e)` would catch all items of the queue and bubble the error to a potentially wrong render call.

But you're right. My initial try/catch block was wrongly set. I've set it inside the queueTask cb function.